src/mlpack/methods/ann/layer/batch_norm_impl.hpp

```diff
@@ -63,68 +72,229 @@ void BatchNorm<InputDataType, OutputDataType>::Reset()
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void BatchNorm<InputDataType, OutputDataType>::Forward(
-    const arma::Mat<eT>& input, arma::Mat<eT>& output)
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
 {
-  // Mean and variance over the entire training set will be used to compute
-  // the forward pass when deterministic is set to true.
-  if (deterministic)
-  {
-    // Normalize the input and scale and shift the output.
-    output = input.each_col() - runningMean;
-    output.each_col() %= gamma / arma::sqrt(runningVariance / count + eps);
-    output.each_col() += beta;
-  }
-  else
+  const size_t batchSize = input.n_cols;
+
+  // Set size of output equal to the size of input.
+  output.set_size(arma::size(input));
+
+  // We will calculate mini batch norm on each channel / feature map.
```

Comment:
```suggestion
  // We will calculate mini-batch norm on each channel / feature map.
```

I think `minibatch` works as well?